app/scripts/lib/util.js

```diff
@@ -173,13 +162,47 @@ function isPrefixedFormattedHexString(value) {
   return /^0x[1-9a-f]+[0-9a-f]*$/iu.test(value)
 }
 
+/**
+ * Returns string with '0x' hex prefix
+ *
+ * @param {string} str
+ * @returns {string}
+ */
+const addHexPrefix = (str) => {
+  if (
+    typeof str !== 'string' ||
+    str.toLowerCase().startsWith('-0x') ||
+    str.toLowerCase().startsWith('0x')
+  ) {
+    return str
```

Comment:
Shouldn't we return `str.toLowerCase()` here? Or, at least ensure that the prefix is `-0x`/`0x` and not `-0X`/`0X`?

Reply:
Makes sense 👍